### ISSUE TITLE
Add `DeviceAuthEndpoint` input field for AzureAD custom endpoints

### ISSUE THREAD

**Issue by enrichman** [rancher/rancher#44778]
Issue:
 - https://github.com/rancher/rancher/issues/42939
 
SURE-4275

**Is your feature request related to a problem? Please describe.**

To completely support the https://github.com/rancher/rancher/pull/44285 we also need to add one extra input field in the AzureAD configuration section, for the custom endpoints.

A user can provide custom endpoints, and we should support also a custom `DeviceAuthEndpoint`.

**Comment by samjustus** [comment]
@gaktive @nwmac  we need this to complete our CLI feature for 2.9, and the June Patch release. The PoC is linked above for your team's reference of the backend changes. Feel free to reach out to Enrico for anything

cc @CamrynCarter for prioritization confirmation

**Comment by gaktive** [comment]
https://github.com/rancher/dashboard/issues/10616 filed to investigate for the UI portion of this, though I'll have to check if this was on our radar before.

**Comment by CamrynCarter** [comment]
@samjustus We need this prioritized from a UI standpoint?

**Comment by samjustus** [comment]
@CamrynCarter correct, for us to complete the feature in full

**Comment by eva-vashkevich** [comment]
Hi @enrichman, does this change affect Steve or Norman API? UI is using /v3/authconfig to create and manage auth configuration including AzureAD

Thank you,
Evgeniya

**Comment by enrichman** [comment]
@eva-vashkevich yes, it extends the API adding the `deviceAuthEndpoint` extra field to handle the new DeviceAuthEndpoint, so no breaking changes.

**Comment by samjustus** [comment]
UI created their own issue for completing this work https://github.com/rancher/dashboard/issues/10616
